File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
bool cmp(vector<int>& a, vector<int>& b) {
    return a[1] < b[1]; 
}
class Solution {

public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
      sort(intervals.begin(), intervals.end(), cmp); 
       int ans=0,end = intervals[0][1];
        for(int i=1;i<intervals.size();i++){
            if(intervals[i][0] < end){
                ans++;
            } 
            else end = intervals[i][1];

        }
        return ans;
    }
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        
       Arrays.sort(intervals,(a,b)->Integer.compare(a[1],b[1]));
       int end = intervals[0][1],cnt=0;
        for(int i=1;i<intervals.length;i++){
              if(intervals[i][0] < end){
                cnt++;
                continue;
              }
               end = intervals[i][1];
        }
        return cnt;
    }
}